drivers/adc/Kconfig.ti_adc108s102

```diff
@@ -0,0 +1,43 @@
+# Kconfig - ADC configuration options
+
+#
+# Copyright (c) 2015 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig ADC_TI_ADC108S102
+	bool "TI adc108s102 chip driver"
+	select SPI
```

Comment:
It needs to depend on something. It should not appear in menuconfig for platforms that do not have it.

Reply:
should it depend on SPI instead of selecting it?

Reply:
@galak No selecting SPI is ok. It should depends on some more general. For example `ADC_QMSI` depends on `QMSI.`
This ADC is external chip, it is not some ADC built in any SoC. Probably @nashif knows how support for features that comes from external chips should be configured in Kconfig.
Right now, when I use menuconfig I can see that I can enable `ADC_TI_ADC108S102` for nRF5 evaluation kits but they do not even have such chip on PCB.

Reply:
I'm not sure I agree that it should select SPI, but why does it need to depend on anything?  Will the fact that its under the ADC menuconfig not be sufficient?

Reply:
@galak Maybe it should not select SPI, however it is not the case here.
Is not it weird that when you run `make BOARD=xxx menuconfig` you can configure some stuff that is not related to that BOARD at all?

Reply:
I don't think its weird for most of the boards have various headers that someone might wire something up on.  So who says someone doesn't connect the SPI pins on board to a TI_ADC108S102?

Reply:
@galak Shouldn't there be, in that situation, a more general config that would indicate that TI_ADC108S102 is on board?
About `select SPI` it depends on what rules we follow. Should higher level drivers automatically enable lower level drivers that they utilize or should we first enable lower level drivers to be able to configure the higher ones?

Reply:
Having one Kconfig where user would be able to enable all external chips that are on board would keep menuconfig more organized, I guess. Lets suppose there is support for multiple external chips, then menuconfig would consist mostly of irrelevant configs.
For example there would be:
```
config TI_ADC108S102
bool "Enable suuport for TI_ADC108S102"
help
  Setting this config add support for TI_ADC108S102 chip, set it if it is on your board
```

Add based upon that TI_ADC108S102 would appear in ADC menu.
What do you think @galak @nashif?

Reply:
if I am not mistaken, selecting SPI was originally int he code and is not part of this change, we should not make any changes here and if select SPI is not needed that can be addressed in a different patch. I am ok with the split as is and as long it does not introduce any additional changes beside the split.